src/main/java/org/jalasoft/moi/controller/endpoints/FileController.java

```diff
@@ -105,9 +112,24 @@ public void deleteFileById(@PathVariable Long id) {
      * @return the output from the execution
      */
     @PostMapping(path = "/execute")
-    public String executeCode(@RequestParam(value = "Project Id") Long projectID){
+    public String executeCode(@RequestParam(value = "Project Id") Long projectID) throws IOException {
```

Comment:
It's an endpoint, I believe you should not throw an exception in the signature of the method.